### PR TITLE
update mlkitdownloader to use executors

### PR DESCRIPTION
**Pull request by VinayGuthal**

### CODE REVIEW

File: firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/CustomModelDownloadService.java [hunk 1]
```diff
@@ -105,12 +104,13 @@ public CustomModelDownloadService(
       FirebaseOptions options,
       Provider<FirebaseInstallationsApi> installationsApi,
       FirebaseMlLogger eventLogger,
-      CustomModel.Factory modelFactory) {
+      CustomModel.Factory modelFactory,
+      Executor blockingExecutor) {
```
Comment (vkryachko) [inline firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/CustomModelDownloadService.java:108]: add `@Blocking` here and remove suppress lint

File: firebase-common/src/main/java/com/google/firebase/platforminfo/GlobalLibraryVersionRegistrar.java [hunk 2]
```diff
@@ -37,7 +39,9 @@ public class GlobalLibraryVersionRegistrar {
    * <p>It is the responsibility of the caller to register the version at app launch.
    */
   public void registerVersion(String sdkName, String version) {
+
     synchronized (infos) {
+      LOGGER.info("Registering version");
```
Comment (vkryachko) [inline firebase-common/src/main/java/com/google/firebase/platforminfo/GlobalLibraryVersionRegistrar.java:44]: intended?

File: firebase-ml-modeldownloader/src/androidTest/java/com/google/firebase/ml/modeldownloader/TestPublicApi.java [hunk 3]
```diff
@@ -63,16 +63,16 @@ public void listModels() throws ExecutionException, InterruptedException {
...
-    Tasks.await(modelTask);
-    assertTrue(modelTask.isSuccessful());
+//    Task<CustomModel> modelTask =
+//        FirebaseModelDownloader.getInstance()
+//            .getModel(
+//                MODEL_NAME_LOCAL,
+//                DownloadType.LATEST_MODEL,
+//                new CustomModelDownloadConditions.Builder().build());
+//    Tasks.await(modelTask);
+//    assertTrue(modelTask.isSuccessful());
```
Comment (vkryachko) [inline firebase-ml-modeldownloader/src/androidTest/java/com/google/firebase/ml/modeldownloader/TestPublicApi.java:73]: why is it ok to comment out this test?

File: firebase-common/src/main/java/com/google/firebase/platforminfo/GlobalLibraryVersionRegistrar.java [hunk 4]
```diff
@@ -37,6 +37,7 @@ public class GlobalLibraryVersionRegistrar {
    * <p>It is the responsibility of the caller to register the version at app launch.
    */
   public void registerVersion(String sdkName, String version) {
+
```
Comment (vkryachko) [inline firebase-common/src/main/java/com/google/firebase/platforminfo/GlobalLibraryVersionRegistrar.java:40]: remove?

File: firebase-ml-modeldownloader/src/test/java/com/google/firebase/ml/modeldownloader/FirebaseModelDownloaderTest.java [hunk 5]
```diff
@@ -138,6 +138,7 @@ public void setUp() throws Exception {
             mockFileManager,
             mockEventLogger,
             executor,
+            executor,
```
Comment (vkryachko) [inline firebase-ml-modeldownloader/src/test/java/com/google/firebase/ml/modeldownloader/FirebaseModelDownloaderTest.java:141]: pls use appropriate `TestOnlyExecutors` here instead.

Comment (rlazo) [reply firebase-ml-modeldownloader/src/test/java/com/google/firebase/ml/modeldownloader/FirebaseModelDownloaderTest.java:141]: +1

**Review by rlazo — APPROVED** [review]

**Pull request merged**